addons/pos_restaurant/static/src/app/floor_screen/floor_screen.js

```diff
@@ -361,6 +361,8 @@ export class FloorScreen extends Component {
         if (table) {
             await this.pos.unsetTable();
         }
+        // Set order to null when reaching the floor screen.
+        this.pos.set_order(null);
```

Comment:
This is moved outside `unsetTable` so that whenever we reach the floorscreen, we make sure no order is selected.